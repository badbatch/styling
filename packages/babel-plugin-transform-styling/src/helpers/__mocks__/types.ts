import { StylingNamedExports } from "../../types";

export interface MockBuildTransformedFile {
  _setFile: (file: string) => void;
  default: (namedExports: StylingNamedExports) => string;
}

export interface MockEvalStylingFile {
  _setFile: (file: StylingNamedExports) => void;
  default: (filename: string) => StylingNamedExports;
}
