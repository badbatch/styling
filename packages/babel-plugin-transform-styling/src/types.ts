import { NodePath, Visitor } from "@babel/traverse";
import { StringObject } from "@styling/types";

export type ExportsArgsMap = Map<string, { type: string; value: string }>;

export interface ExportsArgsResult {
  identifiers: string[];
  map: ExportsArgsMap;
}

export interface PluginState {
  cwd: string;
  file: {
    path: NodePath;
  };
  filename: string;
}

export interface PluginResult {
  visitor: Visitor<PluginState>;
}

export interface StylingNamedExport {
  propsToClassNamesMap: StringObject;
}

export interface StylingNamedExports {
  [key: string]: StylingNamedExport;
}
