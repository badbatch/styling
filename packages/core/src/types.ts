import { ComponentType, ReactHTML, ReactNode, ReactSVG } from "react";

export interface ForwardedProps {
  as?: ComponentType | keyof ReactHTML | keyof ReactSVG;
  children?: ReactNode;
  className?: string;
  [key: string]: any; // tslint:disable-line no-any
}

export interface ReturnedElementProps {
  children?: ReactNode;
  className?: string;
  ref?: React.Ref<any>; // tslint:disable-line no-any
  [key: string]: any; // tslint:disable-line no-any
}
