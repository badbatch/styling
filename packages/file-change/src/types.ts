import { PlainObject } from "@styling/types";

export interface CheckFileAndImportsOptions {
  lastCheckedFiles: PlainObject;
  parentLastCheckedTimestamp?: number;
}
