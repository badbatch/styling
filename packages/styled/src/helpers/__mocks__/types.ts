export interface MockBuildBaseSelector {
  _setSelector: (selector: string) => void;
  default: (componentName: string, prefix?: string) => string;
}
