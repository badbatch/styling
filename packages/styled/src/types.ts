import { CSSObject } from "postcss-js";

export interface Metadata {
  componentName: string;
  sourceDir: string;
}

export interface MockBuildBaseSelector {
  _setSelector: (selector: string) => void;
  default: (componentName: string, prefix?: string) => string;
}

export interface PropKeyComboCSS {
  [key: string]: {
    css: CSSObject;
    keyCombo: string[];
    selector: string;
  };
}

export type PropListExact<P extends {}> = Array<keyof P | [keyof P, string[]] | [keyof P, string | number]>;
