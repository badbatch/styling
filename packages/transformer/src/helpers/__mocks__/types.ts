import { StylingNamedExports } from "../../types";

export interface MockBuildTransformedFile {
  _setFile: (file: string) => void;
  default: (namedExports: StylingNamedExports) => string;
}

export interface MockLoadStylingFile {
  _setFile: (file: StylingNamedExports) => void;
  default: (filename: string) => StylingNamedExports;
}
