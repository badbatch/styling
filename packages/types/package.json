{
  "name": "@styling/types",
  "version": "0.0.6",
  "description": "The Styling types module.",
  "keywords": [
    "css-in-js",
    "react",
    "styled components",
    "stylesheet",
    "styling"
  ],
  "homepage": "https://github.com/badbatch/styling",
  "bugs": {
    "url": "https://github.com/badbatch/styling/issues"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/badbatch/styling"
  },
  "license": "MIT",
  "author": "Dylan Aubrey <dylanaubrey@gmail.com>",
  "main": "lib/main/index.js",
  "types": "lib/types/index.d.ts",
  "scripts": {
    "compile": "concurrently npm:compile:*",
    "compile-watch": "concurrently npm:compile-watch:*",
    "compile-watch:main": "yarn run compile:main --watch",
    "compile-watch:types": "yarn run compile:types --watch",
    "compile:main": "cross-env BABEL_ENV=main babel ./src --out-dir ./lib/main --extensions \".ts\" --source-maps --config-file ../../babel.config.js",
    "compile:types": "tsc --build"
  },
  "dependencies": {},
  "devDependencies": {},
  "peerDependencies": {
    "@babel/runtime": "^7.0.0",
    "core-js": "> 4.0",
    "lodash": "^4.17.5"
  },
  "publishConfig": {
    "access": "public"
  }
}