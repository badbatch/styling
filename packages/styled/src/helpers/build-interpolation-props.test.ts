import { CSSVariablePropList } from "@styling/types";
import buildInterpolationProps from "./build-interpolation-props";

describe("buildInterpolationProps", () => {
  it("SHOULD return the correct object properties and values", () => {
    const propKeyCombo = ["active", "disabled", "error", "selected", "success", "status::approved"];

    const cssVariablePropList: CSSVariablePropList = [
      ["maxHeight", "100%"],
      ["maxWidth", "450px"],
    ];

    expect(buildInterpolationProps(propKeyCombo, cssVariablePropList)).toMatchSnapshot();
  });
});
