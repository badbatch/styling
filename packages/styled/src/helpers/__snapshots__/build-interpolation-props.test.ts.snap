// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildInterpolationProps SHOULD return the correct object properties and values 1`] = `
Object {
  "active": true,
  "disabled": true,
  "error": true,
  "maxHeight": "var(--max-height, 100%)",
  "maxWidth": "var(--max-width, 450px)",
  "selected": true,
  "status": "approved",
  "success": true,
}
`;
